core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedClusterMessage.scala

```diff
@@ -52,6 +52,8 @@ private[spark] object CoarseGrainedClusterMessages {
   case class UpdateDelegationTokens(tokens: Array[Byte])
     extends CoarseGrainedClusterMessage
 
+  case object DecommissionSelf extends CoarseGrainedClusterMessage // Mark as decommissioned.
```

Comment:
IMHO, the DecommissionSelf naming is a bit ambiguous: "Who is self here" ? The sender or the receiver ? 

This message is now send from the driver to the executor: So perhaps we should just repurpose DecommissionExecutor with a check for the executorId ? 

Not a big deal but trying to reduce the number of message types introduced by this feature ;)

Reply:
I think decommissionself is pretty clearly telling the receiver to decommission itself. That being said I'm open to renaming.

Reply:
Sounds good ;-) I checked that DecommissionSelf is not indeed used anywhere else, so it should be unambiguous. Lets keep the name.